src/ipc/ipc3/helper.c

```diff
@@ -352,6 +352,11 @@ struct comp_dev *comp_new(struct sof_ipc_comp *comp)
 	if (!drv)
 		return NULL;
 
+	/* Hack to fix mismatch between uuid and type as most downstream
+	 * systems match on the type but get_drv will match on uuid if present.
+	 **/
+	comp->type = drv->type;
```

Comment:
wouldn't it be better to check and to error out in case of mismatch?

Reply:
if not possible to error out then due to usersapce, then it could be logged.

Reply:
Looks like regardless of what way I do it, unit tests fail. I am curious what on device testing looks like @wszypelt

Reply:
LNL test appear to pass but unit tests do not.

Here are my thoughts on the two solutions on their ups/downs

| |override|expect match|
|-|------------|--------------------|
|upside|should always work regardless of state of host / topology|requires correctness|
|downside|- we are overriding host information|1.will break any topology that is not set correctly 2.may break with module adapter changes as enum changes with that.|

Reply:
@lgirdwood @lyakh would be great to get clarification on this, this is blocking the fuzzer currently

Reply:
> @lgirdwood @lyakh would be great to get clarification on this, this is blocking the fuzzer currently

@cujomalainey you mean a clarification of unit test failures? Sorry, I don't have an answer off the top of my head, I guess it just needs to be debugged and fixed. Supposedly unit tests use those fields in some different incompatible with our assumptions here way. @singalsu would you have an idea maybe?

Reply:
No, which one is better given both have risks of topology mismatch, if we should do override or check enum + uuid. The double checks i listed above, just curious if there is any thoughts on the risks.

Reply:
@cujomalainey I'm for correctness FWIW